Advanced/Play Asset Delivery/Assets/PlayAssetDelivery/Runtime/AddressablesInitSingleton.cs

```diff
@@ -42,86 +41,116 @@ public bool HasInitialized
-                Setup();
+            {
+                // Core Unity asset packs use install-time delivery and are already installed.
+                StartCoroutine(DownloadCustomAssetPacksData());
+            }
             else
             {
+                // Core Unity asset packs use fast-follow or on-demand delivery and need to be downloaded.
                 string[] coreUnityAssetPackNames = AndroidAssetPacks.GetCoreUnityAssetPackNames(); // only returns names of asset packs that are fast-follow or on-demand delivery
                 if (coreUnityAssetPackNames.Length == 0)
-                    Debug.LogError("Cannot retrieve core Unity asset pack names. PlayCore Plugin is not installed.");
+                    LogWarning("Cannot retrieve core Unity asset pack names. PlayCore Plugin is not installed.", true);
                 else
                     AndroidAssetPacks.DownloadAssetPackAsync(coreUnityAssetPackNames, CheckDownloadStatus);
             }
-#else
-            Setup();
-#endif
         }
 
-        void Setup()
+        void LoadFromEditorData()
         {
-            Addressables.ResourceManager.ResourceProviders.Add(new PlayAssetDeliveryAssetBundleProvider());
-            Addressables.ResourceManager.InternalIdTransformFunc = AssetPackTransformFunc;
-            StartCoroutine(LoadCustomAssetPacksData());
+            if (File.Exists(CustomAssetPackUtility.CustomAssetPacksDataEditorPath))
+            {
+                InitializeBundleToAssetPackMap(File.ReadAllText(CustomAssetPackUtility.CustomAssetPacksDataEditorPath));
+                Addressables.ResourceManager.InternalIdTransformFunc = EditorTransformFunc;
+            }
+            else if (File.Exists(CustomAssetPackUtility.CustomAssetPacksDataRuntimePath))
+            {
+                InitializeBundleToAssetPackMap(File.ReadAllText(CustomAssetPackUtility.CustomAssetPacksDataRuntimePath));
+                Addressables.ResourceManager.InternalIdTransformFunc = EditorTransformFunc;
+            }
+            HasInitialized = true;
         }
 
-        IEnumerator LoadCustomAssetPacksData()
+        void LogWarning(string message, bool finishInitializing)
         {
-            string path = Path.Combine(Application.streamingAssetsPath, "CustomAssetPacksData.json");
-            UnityWebRequest www = UnityWebRequest.Get(path);
+            if (logInitializationWarnings)
+                Debug.LogWarning($"{message} Default internal id locations will be used instead.");
+            HasInitialized = finishInitializing;
+        }
+
+        IEnumerator DownloadCustomAssetPacksData()
+        {
+            UnityWebRequest www = UnityWebRequest.Get(CustomAssetPackUtility.CustomAssetPacksDataRuntimePath);
             yield return www.SendWebRequest();
 
             if (www.result != UnityWebRequest.Result.Success)
+                LogWarning($"Could not load 'CustomAssetPacksData.json' : {www.error}.", true);
```

Comment:
does it make more sense to get the file name from CustomAssetPackUtility.CustomAssetPacksDataRuntimePath instead of the hardcoding in the CustomAssetPacksData.json?